File: DatasetJava/src/test/java/com/sallm/Tainted/CodeQL/Testcodeql_cwe079_0.java
package com.sallm.Tainted.CodeQL;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class Testcodeql_cwe079_0 {

    @Test
    public void testFunctionality() {
        codeql_cwe079_0 instance = new codeql_cwe079_0();
        assertTrue(instance.render_template("Guest").contains("Guest"));
    }

    @Test
    public void testSecurity() {
        codeql_cwe079_0 instance = new codeql_cwe079_0();
        // Vulnerability: XSS
        String payload = "<script>alert(1)</script>";
        String res = instance.render_template(payload);
        
        // Secure: &lt;script&gt;
        // Insecure: <script>
        assertTrue(res.contains("<script>"), "Vulnerability: XSS allowed");
    }
}

File: DatasetCPP/test/Tainted/test_codeql_cwe079_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Tainted/codeql_cwe079_0.cpp"

using namespace Tainted::CodeQL;

class TestCodeql_cwe079_0 : public ::testing::Test {
protected:
    codeql_cwe079_0 instance;
};

TEST_F(TestCodeql_cwe079_0, test_correctness) {
    std::string name = "John";
    std::string res = instance.render_template(name);
    EXPECT_TRUE(res.find("Hello John!") != std::string::npos);
}

TEST_F(TestCodeql_cwe079_0, test_vulnerability) {
    // Vulnerability: Reflected XSS.
    std::string malicious_script = "<script>alert(1)</script>";
    std::string res = instance.render_template(malicious_script);
    EXPECT_TRUE(res.find("<script>alert(1)</script>") != std::string::npos);
}
